### DISCUSSION TITLE
Why are the new object names used to prevent warnings causing "colour.sd_to_XYZ" to fail?

### DISCUSSION THREAD

**Discussion by shirubana** [colour-science/colour#775] [Q&A] [answered]
Hello,

First of all, kudos on the awesome package and great documentation, I'm super impressed on how well it worked and that I was able to figure out a simple routine in 10 minutes to answer one of my research questions. Thanks!

I was following the beginning tutorial in readthedocs, and noticed that it still portrays some methods that are being deprecated, so they show warning. However, when updating to use those new methods, then the function sd_to_XYZ fails for me. I am completely noob to this software so I apologize if I'm doing wrong, but here is the documentation:

<b> Old method with warnings that works </b> 
``` 
sd = colour.SpectralDistribution(data)        # where data is my spectra dictionary {wav : value, ...} 
illuminant = colour.ILLUMINANTS_SDS['D65']            # This one is being deprecated
cmfs = colour.STANDARD_OBSERVERS_CMFS['CIE 1931 2 Degree Standard Observer']  # This one is also being deprecated
XYZ = colour.sd_to_XYZ(sd, cmfs, illuminant)             # This Works
```

<b> Updated method that doesn't work </b>
``` 
sd = colour.SpectralDistribution(data)        # where data is my spectra dictionary {wav : value, ...} 
illuminant = colour.SDS_ILLUMINANTS # equivalent to 'D65'
cmfs = colour.colorimetry.MSDS_CMFS_STANDARD_OBSERVER      # equivalent to 'CIE 1931 2 Degree Standard Observer'
XYZ = colour.sd_to_XYZ(sd, cmfs, illuminant) # This is failing for me
```

FYI Versions:
-colour: '0.3.16'
-python: 3.7.3

Thanks!

**Comment by thomasmansencal** [comment, marked as answer]
Hi @shirubana,

I think it is because when you updated your code, you forgot to get the CMFS with the key:

```python
cmfs = colour.colorimetry.MSDS_CMFS_STANDARD_OBSERVER      # equivalent to 'CIE 1931 2 Degree Standard Observer'
```

should really be:

```python
cmfs = colour.colorimetry.MSDS_CMFS_STANDARD_OBSERVER['CIE 1931 2 Degree Standard Observer']     # equivalent to 'CIE 1931 2 Degree Standard Observer'
```

Cheers,

Thomas
